### DISCUSSION TITLE
Why using `meson`?

### DISCUSSION THREAD

**Discussion by TornaxO7** [flxzt/rnote#324] [Q&A] [answered]
Hi, I'm wondering why `rnote` uses an external build-system (in this case `meson`) to build this project. Since it's using rust, why not just use `cargo` then?

**Comment by 0xMRTT** [comment]
Because it's using gnome and gtk stuff. And gnome use meson.

**Comment by flxzt** [comment, marked as answer]
It would be possible just to use cargos `build.rs` script, but meson is a lot more ergonomic.  Checking dependencies, configuring / installing files and valdiating them is a lot easier with meson.

As already mentioned, it is also somewhat of a convention with gnome apps and I was using [this project template](https://gitlab.gnome.org/World/Rust/gtk-rust-template) to get started. Meson also integrates nicely into flatpak manifests.
